source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionWillAttemptACancelCallWhenGetStatusFails.cs

```diff
@@ -0,0 +1,87 @@
+                .WithRetryDuration(TimeSpan.FromSeconds(10))
+                .WithHalibutTimeoutsAndLimits(new HalibutTimeoutsAndLimits
+                {
+                    RetryCountLimit = 1,
+                    ConnectionErrorRetryTimeout = TimeSpan.FromSeconds(1),
+                    RetryListeningSleepInterval = TimeSpan.FromSeconds(1),
+                    PollingRequestQueueTimeout = TimeSpan.FromSeconds(5)
+                })
+                .WithResponseMessageTcpKiller(out var responseMessageTcpKiller)
+                .WithTentacleServiceDecorator(new TentacleServiceDecoratorBuilder()
+                    .RecordMethodUsages(tentacleConfigurationTestCase, out var recordedUsages)
+                    .DecorateAllScriptServicesWith(u => u
+                        .BeforeGetStatus(
+                            async () =>
+                            {
+                                await Task.CompletedTask;
+                                responseMessageTcpKiller.KillConnectionOnNextResponse();
+                            })
+                    )
+                    .Build())
+                .Build(CancellationToken);
+
+            var waitForFile = Path.Combine(clientTentacle.TemporaryDirectory.DirectoryPath, "waitforme");
+            var fileCreatedByScript = Path.Combine(clientTentacle.TemporaryDirectory.DirectoryPath, "scriptFile");
+
+            var startScriptCommand = new TestExecuteShellScriptCommandBuilder()
+                .SetScriptBody(new ScriptBuilder()
+                    .Print("hello")
+                    .WaitForFileToExist(waitForFile)
+                    .Print("AllDone")
+                    .CreateFile(fileCreatedByScript))
+                .Build();
+
+            var inMemoryLog = new InMemoryLog();
+
+            var execScriptTask = Task.Run(
+                async () => await clientTentacle.TentacleClient.ExecuteScript(startScriptCommand, CancellationToken, null, inMemoryLog),
+                CancellationToken);
+
+            await Wait.For(() => recordedUsages.For(nameof(IAsyncClientScriptServiceV2.GetStatusAsync)).LastException != null,
+                TimeSpan.FromSeconds(60),
+                () => throw new Exception("GetStatus did not error"),
+                CancellationToken);
+
+            // Assert
+
+            // We should find that the cancel request was sent to tentacle.
+            await Wait.For(() => recordedUsages.For(nameof(IAsyncClientScriptServiceV2.CancelScriptAsync)).Completed > 0,
+                TimeSpan.FromSeconds(60),
+                () => throw new Exception("Cancel script was never called."),
+                CancellationToken);
+
+            // Let the script finish.
+            File.WriteAllText(waitForFile, "");
+            await Task.Delay(TimeSpan.FromSeconds(2)); // It's hard to know if the script was killed or not, if we do the assert below too quickly we will
+            // incorrectly see the script as being cancelled.
```

Comment:
re: the comment, "incorrectly see the script as being cancelled" is confusing. 

I would think that by checking too quickly, the task might still be running when we do the check and therefore it's possible that the task may not have been cancelled properly and will still write the file.

But if we wait 2 seconds, we're giving the task plenty of time to either be cancelled, or completed if something isn't work as expected, _then_ doing the check.

How we can rephrase the comment? I know it's just a comment, but it's a little confusing in an already complex scenario.

Reply:
Yeah that's a big improvement - it is closer to the real world scenario and doesn't involve waiting for some arbitrary amount of time.

Very nicely done.